### PR TITLE
Check for system version of simpleini.

### LINKED ISSUE #4679: Use system library version of simpleini when available
simpleini appears to be featured as a package on a range of systems, so it might be good to support linking with the system version rather than fetching it directly from GitHub.

https://repology.org/project/simpleini/versions

Similar change for reference https://github.com/diasurgical/devilutionX/pull/4464

### PR DESCRIPTION
**Pull request by YuviTz1**
For some reason, the apt package for simpleini does not have ConvertUTF.h and ConvertUTF.c files required by SimpleIni.h. Manually copying these files to /usr/include results in completion of the build process.
Do I need to create ```DEVILUTIONX_SYSTEM_``` flag for simpleini?

Resolves #4679

### CODE REVIEW

**Comment by glebm** [comment]
> Do I need to create `DEVILUTIONX_SYSTEM_` flag for simpleini?

Yes please, all of the packages should be set up the same way

**Review by tsunamistate — CHANGES_REQUESTED** [review]
Would also be good to add `simpleini` to `vcpkg.json` to permit VCPKG handle the system version

File: vcpkg.json [hunk 1]
```diff
@@ -3,7 +3,8 @@
     "version-string": "1.3.0",
     "dependencies": [
         "fmt",
-        "bzip2"
+        "bzip2",
+		"simpleini"
```
Comment (glebm) [inline vcpkg.json:7]: The rest of the lines use spaces for indentation but this line uses a tab.

**Review by glebm — DISMISSED** [review]

**Comment by tsunamistate** [comment]
Tried configuring with Visual Studio 2022.
`vcpkg` does install `simpleini`, but after that, CMake doesn't find it for some reason and uses FetchContent version

```
1> CMake generation started for configuration: 'x64-Debug'.
1> Found and using vcpkg toolchain file (D:/src/vcpkg/scripts/buildsystems/vcpkg.cmake).
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\Ghost\CMakeBuilds\f9797d38-6592-4263-b610-70dd4d4e5ca3\install\x64-Debug" -DDISCORD_INTEGRATION:BOOL="True" -DCMAKE_C_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.32.31326/bin/Hostx86/x64/cl.exe" -DCMAKE_CXX_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.32.31326/bin/Hostx86/x64/cl.exe"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" -DCMAKE_TOOLCHAIN_FILE="D:/src/vcpkg/scripts/buildsystems/vcpkg.cmake" "D:\src\devilutionX" 2>&1"
1> Working directory: D:\src\devilutionX\build\x64-Debug
1> [CMake] -- Running vcpkg install
1> [CMake] Detecting compiler hash for triplet x64-windows...
1> [CMake] The following packages will be built and installed:
1> [CMake]     simpleini[core]:x64-windows -> 4.19
1> [CMake] Restored 0 packages from C:\Users\Ghost\AppData\Local\vcpkg\archives in 725.5 us. Use --debug to see more details.
1> [CMake] Installing 1/1 simpleini:x64-windows...
1> [CMake] Building simpleini[core]:x64-windows...
1> [CMake] -- Downloading https://github.com/brofield/simpleini/archive/9b3ed7ec815997bc8c5b9edf140d6bde653e1458.tar.gz -> brofield-simpleini-9b3ed7ec815997bc8c5b9edf140d6bde653e1458.tar.gz...
1> [CMake] -- Extracting source D:/src/vcpkg/downloads/brofield-simpleini-9b3ed7ec815997bc8c5b9edf140d6bde653e1458.tar.gz
1> [CMake] -- Using source at D:/src/vcpkg/buildtrees/simpleini/src/de653e1458-5521b516e8.clean
1> [CMake] -- Installing: D:/src/vcpkg/packages/simpleini_x64-windows/include/SimpleIni.h
1> [CMake] -- Installing: D:/src/vcpkg/packages/simpleini_x64-windows/include/ConvertUTF.h
1> [CMake] -- Installing: D:/src/vcpkg/packages/simpleini_x64-windows/include/ConvertUTF.c
1> [CMake] -- Installing: D:/src/vcpkg/packages/simpleini_x64-windows/share/simpleini/copyright
1> [CMake] -- Performing post-build validation
1> [CMake] -- Performing post-build validation done
1> [CMake] Stored binary cache: C:\Users\Ghost\AppData\Local\vcpkg\archives\b9\b99e4864c90e0d00c874c6fa6f722aae105d45b485eff8438f3a18478cf17186.zip
1> [CMake] Elapsed time to handle simpleini:x64-windows: 1.86 s
1> [CMake] 
1> [CMake] Total elapsed time: 9.302 s
1> [CMake] 
1> [CMake] The package bzip2 is compatible with built-in CMake targets:
1> [CMake] 
1> [CMake]     find_package(BZip2 REQUIRED)
1> [CMake]     target_link_libraries(main PRIVATE BZip2::BZip2)
1> [CMake] 
1> [CMake] The package fmt provides CMake targets:
1> [CMake] 
1> [CMake]     find_package(fmt CONFIG REQUIRED)
1> [CMake]     target_link_libraries(main PRIVATE fmt::fmt)
1> [CMake] 
1> [CMake]     # Or use the header-only version
1> [CMake]     find_package(fmt CONFIG REQUIRED)
1> [CMake]     target_link_libraries(main PRIVATE fmt::fmt-header-only)
1> [CMake] 
1> [CMake] simpleini is header-only and can be used from CMake via:
1> [CMake] 
1> [CMake]     find_path(SIMPLEINI_INCLUDE_DIRS "ConvertUTF.c")
1> [CMake]     target_include_directories(main PRIVATE ${SIMPLEINI_INCLUDE_DIRS})
1> [CMake] 
1> [CMake] libsodium provides CMake targets:
1> [CMake]     # this is heuristically generated, and may not be correct
1> [CMake]     find_package(unofficial-sodium CONFIG REQUIRED)
1> [CMake]     target_link_libraries(main PRIVATE unofficial-sodium::sodium)
1> [CMake] 
1> [CMake] sdl2 provides CMake targets:
1> [CMake]     # this is heuristically generated, and may not be correct
1> [CMake]     find_package(SDL2 CONFIG REQUIRED)
1> [CMake]     target_link_libraries(main PRIVATE SDL2::SDL2 SDL2::SDL2main)
1> [CMake] 
1> [CMake] sdl2-image provides CMake targets:
1> [CMake]     # this is heuristically generated, and may not be correct
1> [CMake]     find_package(sdl2-image CONFIG REQUIRED)
1> [CMake]     target_link_libraries(main PRIVATE SDL2::SDL2_image)
1> [CMake] 
1> [CMake] gtest provides CMake targets:
1> [CMake]     # this is heuristically generated, and may not be correct
1> [CMake]     find_package(GTest CONFIG REQUIRED)
1> [CMake]     target_link_libraries(main PRIVATE GTest::gmock GTest::gtest GTest::gmock_main GTest::gtest_main)
1> [CMake] 
1> [CMake] -- Running vcpkg install - done
1> [CMake] -- 📚 zlib: dynamic system library
1> [CMake] -- 📚 bzip2: dynamic system library
1> [CMake] -- 📚 SDL2: dynamic system library
1> [CMake] -- 📚 SDL_image: dynamic system library
1> [CMake] -- 📚 SDL_audiolib: static library from source
1> [CMake] -- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
1> [CMake] -- 📚 libsodium: dynamic system library
1> [CMake] -- 📚 libfmt: dynamic system library
1> [CMake] -- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
1> [CMake] -- Suitable system simpleini package not found, will use simpleini from source
1> [CMake] -- 📚 simpleini: static library from source
1> [CMake] -- 📚 googletest: dynamic system library
1> [CMake] -- Configuring done
1> [CMake] -- Generating done
1> [CMake] -- Build files have been written to: D:/src/devilutionX/build/x64-Debug
1> Extracted CMake variables.
1> Extracted source files and headers.
1> Extracted code model.
1> Extracted toolchain configurations.
1> Extracted includes paths.
1> CMake generation finished.
```

@YuviTz1 looks like you're using wrong logic to find simpleini
```
simpleini is header-only and can be used from CMake via:
   find_path(SIMPLEINI_INCLUDE_DIRS "ConvertUTF.c")
   target_include_directories(main PRIVATE ${SIMPLEINI_INCLUDE_DIRS})
```

**Comment by YuviTz1** [comment]
CMake did seem to find it on my Ubuntu based Linux distro
I didn't try with Visual Studio
> @YuviTz1 looks like you're using wrong logic to find simpleini

I'll look into it

**Comment by glebm** [comment]
This did break on my Ubuntu system with simpleini installed, complaining about the lack of ConvertUTF

**Comment by glebm** [comment]
Looking at documentation, the correct fix for our use-case appears to be to define `SI_NO_CONVERSION` before including simpleini! :tada:

**Comment by glebm** [comment]
However, that fix is not available in older versions of SimpleIni, like the one on my Ubuntu 22.04

**Comment by glebm** [comment]
Working on a PR

**Pull request merged**